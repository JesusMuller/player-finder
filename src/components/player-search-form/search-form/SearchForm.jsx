import React from 'react';
import { func, string } from 'prop-types';
import { positions } from '../../../constants/constants'
import TextInput from '../../text-input';
import ComboBox from '../../combo-box';
import SubmitButton from '../../submit-button';
import './SearchForm.scss';

const SearchForm = ({ onSubmitHandler,
    playerNameInput,
    onNameInputChangeHandler,
    onPositionInputChangeHandler,
    playerAgeInput,
    onAgeInputChangeHandler,
    playerNameError,
    playerAgeError }) => (
        <form onSubmit={onSubmitHandler}>
            <div className="input-container">
                <TextInput placeHolder="Player Name"
                    inputValue={playerNameInput}
                    onChangeHandler={onNameInputChangeHandler}
                    errorMessage={playerNameError} />
            </div>
            <div className="input-container">
                <ComboBox optionsInitializer="Position" options={positions}
                    onChangeHandler={onPositionInputChangeHandler} />
            </div>
            <div className="input-container">
                <TextInput placeHolder="Age"
                    inputValue={playerAgeInput}
                    onChangeHandler={onAgeInputChangeHandler}
                    errorMessage={playerAgeError} />
            </div>
            <div className="input-container">
                <SubmitButton buttonText="Search" />
            </div>
        </form>
    );

SearchForm.propTypes = {
    onSubmitHandler: func.isRequired,
    playerNameInput: string.isRequired,
    onNameInputChangeHandler: func.isRequired,
    onPositionInputChangeHandler: func.isRequired,
    playerAgeInput: string.isRequired,
    onAgeInputChangeHandler: func.isRequired,
    playerNameError: string,
    playerAgeError: string
};

export default SearchForm;
