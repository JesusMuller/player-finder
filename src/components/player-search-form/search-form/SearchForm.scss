form {
    padding: 1rem 0;
    display: flex;
    flex-wrap: wrap;

    @media(max-width: 660px) {
        display: block;
    }

    .input-container {
        flex-grow: 1;
        padding-bottom: 10px;

        >span {
            display: block;
            color: #ff0000;
            font-size: .75em;
        }

        >span::before {
            content: '* '
        }

        input:not([type='submit']),
        select {
            width: 90%;

            @media(max-width: 660px) {
                width: 100%;
            }
        }
    }
}