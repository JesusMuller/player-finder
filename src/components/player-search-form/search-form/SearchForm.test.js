import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

const setProps = {
    onSubmitHandler: jest.fn(),
    playerNameInput: '',
    onNameInputChangeHandler: jest.fn(),
    onPositionInputChangeHandler: jest.fn(),
    playerAgeInput: '',
    onAgeInputChangeHandler: jest.fn()
};

describe('<SearchForm />', () => {
    it('Should match <SearchForm /> Snapshot', () => {
        const wrapper = shallow(<SearchForm {...setProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
