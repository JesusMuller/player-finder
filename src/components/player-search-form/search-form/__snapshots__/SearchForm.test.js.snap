// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> Should match <SearchForm /> Snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="input-container"
  >
    <TextInput
      inputValue=""
      onChangeHandler={[MockFunction]}
      placeHolder="Player Name"
    />
  </div>
  <div
    className="input-container"
  >
    <ComboBox
      onChangeHandler={[MockFunction]}
      options={
        Array [
          "Attacking Midfield",
          "Central Midfield",
          "Centre-Back",
          "Centre-Forward",
          "Defensive Midfield",
          "Keeper",
          "Left Midfield",
          "Left Wing",
          "Left-Back",
          "Right-Back",
        ]
      }
      optionsInitializer="Position"
    />
  </div>
  <div
    className="input-container"
  >
    <TextInput
      inputValue=""
      onChangeHandler={[MockFunction]}
      placeHolder="Age"
    />
  </div>
  <div
    className="input-container"
  >
    <SubmitButton
      buttonText="Search"
    />
  </div>
</form>
`;
